src/Microsoft.IdentityModel.Tokens/TokenValidationResult.cs

```diff
@@ -22,6 +22,11 @@ public class TokenValidationResult
         private TokenValidationParameters _validationParameters;
         private TokenHandler _tokenHandler;
 
+        /// <summary>
+        /// The <see cref="SecurityToken"/> to be returned when a validation fails.
+        /// </summary>
+        public SecurityToken TokenOnFailedValidation { get; internal set; }
+
```

Comment:
Consider moving TokenOnFailedValidation property to where SecurityToken property is in TokenValidationResult class.